server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -456,12 +399,13 @@ public static void registerRequestHandler(TransportService transportService, Sea
             SearchFreeContextResponse::readFrom
         );
 
-        transportService.registerRequestHandler(
-            FREE_CONTEXT_ACTION_NAME,
-            freeContextExecutor,
-            SearchFreeContextRequest::new,
-            freeContextHandler
-        );
+        // TODO: remove this handler once the lowest compatible version stops using it
+        transportService.registerRequestHandler(FREE_CONTEXT_ACTION_NAME, freeContextExecutor, in -> {
+            var res = new ScrollFreeContextRequest(in);
```

Comment:
We should do a follow-up to rename the request to just `FreeContext` IMO, but I figured it would be really confusing diff-wise to do a rename and removal via custom deserialization like that in a single commit.